realm/realm-library/src/objectServer/java/io/realm/SyncConfiguration.java

```diff
@@ -183,6 +184,64 @@ public static RealmConfiguration forRecovery(String canonicalPath, @Nullable byt
         return forRecovery(canonicalPath, encryptionKey, schemaMediator);
     }
 
+    /**
+     * Creates an automatic default configuration based on the the currently logged in user.
+     * <p>
+     * This configuration will point to the default Realm on the server where the user was
+     * authenticated.
+     *
+     * @throws IllegalStateException if no user are logged in, or multiple users have. Only one should
+     * be logged in when calling this method.
+     * @return The constructed {@link SyncConfiguration}.
+     */
+    public static SyncConfiguration automatic() {
```

Comment:
I'm not sure why do not call this `default()` instead. Somehow that feels more natural to me.

Reply:
I'm not sure about the default vs automatic naming. To me "default" feels a little more static, while "automatic" is more active. Given that the returned configuration can change between calls due to outside actions (changing the active user), the more active term felt like a better fit. This perception is very subjective, though.

One thing worth noting is that Cocoa's `SyncConfiguration.automatic()` returns a `Realm.Configuration`, not a `SyncConfiguration`. That was done to minimize the amount of code required to use the returned configuration.

Reply:
In Java `SyncConfiguration` is a subclass of `RealmConfiguration` so it can be used everywhere a RealmConfiguration is used.

Good point about `default` feeling more static vs. `automatic` that feels more active. Lets just stick to `automatic()`